legal-api/src/legal_api/services/filings/validations/annual_report.py

```diff
@@ -32,6 +31,28 @@ def requires_agm(business: Business) -> bool:
     return business.legal_type in agm_arr
 
 
+def get_ar_dates(business: Business, start_date, next_ar_year):
+    """Get ar min and max date for the specific year."""
+    check_agm = requires_agm(business)
+    ar_min_date = datetime(next_ar_year, 1, 1).date()
+
+    # Make sure min date is greater than or equal to last_ar_date or founding_date
+    if ar_min_date < start_date:
+        ar_min_date = start_date
+
+    ar_max_date = datetime(next_ar_year, 12, 31).date()
```

Comment:
this ar_max_date will need to be decided differently for extensions and is what I was talking about in the RFC. I think we need to have a design discussion with Thor and Lekshmi on what we'll add for ARs to make it work for extensions